x/distribution/keeper/keeper.go

```diff
@@ -99,6 +99,14 @@ func (k Keeper) WithdrawDelegationRewards(ctx sdk.Context, delAddr sdk.AccAddres
 		return nil, err
 	}
 
+	if rewards.IsZero() {
```

Comment:
A zero coin technically isn't valid. I don't have strong feeling here though in terms of emitting the event. I'd have to defer to others.

Reply:
`IsZero` doubles as an `IsEmpty` check so 0 rewards == no rewards here 🤷🏻‍♂️

but yeah just to clarify, without the added check, the `value` simply gets axed from the event

Reply:
Totally. I meant in the state-machine we do not allow zero coins. However, since this is for UX/client purposes only, I suppose it's OK.